tuf/api/metadata.py

```diff
@@ -939,11 +1015,24 @@ def __init__(
         self.hashes = hashes
         self.unrecognized_fields = unrecognized_fields or {}
 
+    @property
+    def custom(self):
+        if self.unrecognized_fields is None:
+            return None
+        return self.unrecognized_fields.get("custom", None)
+
     @classmethod
     def from_dict(cls, target_dict: Dict[str, Any]) -> "TargetFile":
         """Creates TargetFile object from its dict representation."""
         length = target_dict.pop("length")
         hashes = target_dict.pop("hashes")
+
+        # Do some basic validation checks
+        if length <= 0:
+            raise ValueError(f"Targetfile length must be > 0, got {length}")
+        if not hashes:
+            raise ValueError("Missing targetfile hashes")
```

Comment:
this is actually a bit misleading as None is already handled (pop() will raise), so this actually handles the "empty hashes dict" case... but since the end result is correct and since we have an issue on properly validating hashes, let's just go with this.

Reply:
It was meant to handle existing but empty dictionary:
`hashes: {}`